[Wu Lezheng] iP

Wu-Lezheng opened the pull request:
# Duke
> “Your mind is for having ideas, not holding them.” – David Allen

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do *is*,
1. download it from [here](https://github.com/Wu-Lezheng/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :clown_face:

And it is **FREE!**

Features:
- [x] Managing to-dos
- [x] Managing events
- [x] Managing deadlines
- [ ] Reminders (coming soon)

___

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Wu-Lezheng committed:
Greet, Echo, Exit added

Wu-Lezheng committed:
Add, List completed

Wu-Lezheng committed:
mark function completed

Wu-Lezheng committed:
todo, event, deadline completed

Wu-Lezheng committed:
error handling completed

Wu-Lezheng committed:
delete completed

Wu-Lezheng committed:
clean up some code

Wu-Lezheng committed:
text ui texting completed

Wu-Lezheng committed:
added some more tests

Wu-Lezheng committed:
Correct some indentation

Wu-Lezheng committed:
Add Level 7.Save

This feature save and load data to and from the local disk

Wu-Lezheng committed:
Add Level 8. Dates and Times

This feature allows Duke to understand date and time.

Wu-Lezheng committed:
Merge branch 'branch-Level-7'

Wu-Lezheng committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

Wu-Lezheng committed:
Clean up some code

Wu-Lezheng committed:
Bug fix

This bug fix resolves the problem of date format parsing.

Wu-Lezheng committed:
Add more OOP and include packages

There are a few new features included:
 * The command class and its subclasses
 * A parser class
 * A TaskList class
 * A Ui class
 * Put classes into packages

Wu-Lezheng committed:
Ignore duke.txt

Wu-Lezheng committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

Wu-Lezheng committed:
Add Gradle to the project

Wu-Lezheng committed:
Add JUnit tests

Include some JUnit tests for task package and parser pacckage.

Wu-Lezheng committed:
Include JavaDoc

Wu-Lezheng committed:
Make changes according to the coding standard

Wu-Lezheng committed:
Add list command

Wu-Lezheng committed:
Merge branch 'branch-A-JavaDoc'

Wu-Lezheng committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/task/Task.java

Wu-Lezheng committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/task/Task.java

jjtan444 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,184 @@
+
+        String commandWord;
+        String description;
+        if (fullCommand.contains(" ")) {
```
I like your control flow implementation using spaces.

jjtan444 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,184 @@
+
+        switch (commandWord) {
+        case AddCommand.COMMAND_WORD_DEADLINE:
+            // Fallthrough
```
This seems pretty unique. Is this the best way to do it? 🤔

marclzh commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,65 @@
+     * @param str description of the task.
+     */
+    public Task(String str) {
+        this.description = str;
```
It should be fine to omit `this.`, since there is no ambiguity of what is being assigned here, similarly for the statement below.

jjtan444 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,184 @@
+                return new AddCommand(new Event(message, date));
+            } else {
+                throw new DukeException(
+                        "☹ Please follow the format <event description /at date>");
```
I like how your error message specifies the correct format to the user.

jjtan444 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+     */
+    public void appendToFile(String textToAppend, Ui ui) {
+        try {
+            FileWriter fw = new FileWriter(filePath, true);
```
Perhaps you could use the name `fileWriter` to provide more clarity.

jjtan444 commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,127 @@
+    public List<Task> load() throws DukeException {
+        try {
+            File file = new File(filePath);
+            Scanner s = new Scanner(file);
```
`scanner` might be a better choice of name.

jjtan444 reviewed (commented):
Overall well done, pretty clean code, just need some tweaks to the naming of variables.

marclzh reviewed (commented):
Overall looks well done with no real issues. There are some design implementations which I can learn from as well!

Wu-Lezheng committed:
Add checkstyle to the project

Wu-Lezheng committed:
Add javaFX

Wu-Lezheng committed:
Correct some coding standard issue

Wu-Lezheng committed:
Add FXML to the project

Wu-Lezheng committed:
Modify the code based on JavaFX

shogun187 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,80 @@
+        }
+    }
+
+    /**
```
Just a suggestion, maybe you could remove these methods if you do not need them?

shogun187 reviewed (commented):
Looks good to me! 👍
I really like how you organized the different classes into different packages, makes the structure look really clean 💯

Wu-Lezheng committed:
Modify the code based on FXML

Wu-Lezheng committed:
Merge branch 'brach-Level-10'

Wu-Lezheng committed:
Modify the code based on code quality standards

There are some violations agaisnt the code quality standards.

Code with good readability makes it easy for others to understand.

Let's,
* make magic strings into expressions
* structure the code more logically
* delete unused methods
* make variable names clearer

Wu-Lezheng committed:
Add java assert statements

Wu-Lezheng committed:
Use streams in duke

TaskList: use loops to iterate through the list and print the result.

The loop can be replaced by the use of stream, which introduces
functiona programming to the code.

Let's,
* Use stream instead of loop
* Get more familiar with stream functions
* Practise functional programming

Wu-Lezheng committed:
Merge pull request #1 from Wu-Lezheng/branch-A-Streams

Use streams in duke

Wu-Lezheng committed:
Merge pull request #3 from Wu-Lezheng/branch-A-CodeQuality

Modify the code based on code quality standards

Wu-Lezheng committed:
Merge branch 'master' into branch-A-Assertions

Wu-Lezheng committed:
Merge pull request #2 from Wu-Lezheng/branch-A-Assertions

Add java assert statements

Wu-Lezheng committed:
Create gradle.yml

Wu-Lezheng committed:
Add the feature to detect duplicate inputs

Wu-Lezheng committed:
Merge branch 'C-DetectDuplicates'

Wu-Lezheng committed:
Improve code quality

Wu-Lezheng committed:
Improve GUI

Wu-Lezheng committed:
Clean up some code

Wu-Lezheng committed:
Merge branch 'branch-A-BetterGui'

Wu-Lezheng committed:
Create a personality for Duke

Wu-Lezheng committed:
Merge branch 'branch-A-Personality'

Wu-Lezheng committed:
Add product name and icon

Wu-Lezheng committed:
Add more error handling

Wu-Lezheng committed:
Merge branch 'branch-A-MoreErrorHandling'

Wu-Lezheng committed:
Add some Junit tests

Wu-Lezheng committed:
Give credit to other authors

Wu-Lezheng committed:
Update gradle.yml

Wu-Lezheng committed:
Merge branch 'branch-issue'

Wu-Lezheng committed:
Add Ui screenshot

Wu-Lezheng committed:
Add the feature to search by date

Wu-Lezheng committed:
Merge branch 'branch-date'

Wu-Lezheng committed:
Update part of the user guide

Wu-Lezheng committed:
Update the user guide

Wu-Lezheng committed:
Update README.md

Wu-Lezheng committed:
Update README.md

Wu-Lezheng committed:
Prepare for release

Wu-Lezheng committed:
Bugfix

Wu-Lezheng committed:
Clean up some code

Wu-Lezheng committed:
Modify the user guide

Wu-Lezheng committed:
Bugfix for corrupted file

Wu-Lezheng committed:
Add welcome screen

Wu-Lezheng committed:
Bugfix for corrupted file

Wu-Lezheng committed:
Bugfix for creating local file

Wu-Lezheng committed:
Bugfix

Wu-Lezheng committed:
Merge branch 'branch-bugfix'

# Conflicts:
#	src/main/java/duke/storage/Storage.java

Wu-Lezheng committed:
Bugfix for mac
